gravatar-quickeditor/src/main/java/com/gravatar/quickeditor/ui/editor/GravatarQuickEditorPage.kt

```diff
@@ -16,29 +18,34 @@ import com.gravatar.quickeditor.ui.oauth.OAuthParams
 
 @Composable
 internal fun GravatarQuickEditorPage(
-    gravatarQuickEditorParams: GravatarQuickEditorParams,
-    onAuthError: () -> Unit = {},
+    appName: String,
+    oAuthParams: OAuthParams,
+    onAvatarSelected: (Uri) -> Unit,
```

Comment:
🤔 I'm thinking about whether `GravatarQuickEditorPage` should have a callback with `AvatarUpdateResult` and extract the `Dismissed` from that class. 

The `Dismissed` state only applies to the `BottomSheet`, and it's not directly related to the avatar update, so maybe a different callback only in the `GravatarQuickEditorBottomSheet` could make sense?

Reply:
>The Dismissed state only applies to the BottomSheet

Yeah, that's why I kept two separate callbacks. But, the dismissed state doesn't only apply to `BottomSheet`, there's no reason we couldn't have a `cancel` button within the Page itself. 


>so maybe a different callback only in the GravatarQuickEditorBottomSheet could make sense?
What do you mean by that?

Reply:
> But, the dismissed state doesn't only apply to BottomSheet, there's no reason we couldn't have a cancel button within the Page itself.

I agree; at some point, we can have a cancel button inside the `GravatarQuickEditorPage,` but that button will cancel the whole QuickEditor, right? If we add a profile edition to the `GravatarQuickEditorPage,` tapping that button will dismiss everything, not just the avatar stuff. That is why I think, the `Dismissed` results belong to `GravatarQuickEditorPage` / `GravatarQuickEditorBottomSheet` or both, but not to the `AvatarUpdateResult`.

What I was trying to say with the different callbacks is the following:

```
public fun GravatarQuickEditorBottomSheet(
    appName: String,
    oAuthParams: OAuthParams,
    onAvatarUpdateResult: (AvatarUpdateResult) -> Unit,
    onDismiss: () -> Unit,
    modalBottomSheetState: SheetState = rememberModalBottomSheetState(skipPartiallyExpanded = true),
) {
```

In fact, it's possible that we need to call several times `onAvatarUpdateResult`, because I tap over different avatars in the QuickEditor and then I manually dismiss the bottomSheet.

Reply:
>In fact, it's possible that we need to call several times onAvatarUpdateResult, because I tap over different avatars in the QuickEditor and then I manually dismiss the bottomSheet.


Oh right! I forgot about that. Thanks!

>That is why I think, the Dismissed results belong to GravatarQuickEditorPage / GravatarQuickEditorBottomSheet or both, but not to the AvatarUpdateResult.

So the idea was that it would be changed to a more generic result later when we have more options than avatar. Something like `QuickEditorResult`. But the above point about multiple invocations makes sense. I will re-work that.

Reply:
Ok, I've modified it to

```
public fun GravatarQuickEditorBottomSheet(
    ...
    onAvatarSelected: (AvatarUpdateResult) -> Unit,
    onDismiss: (error: GravatarQuickEditorError?) -> Unit = {},
)
```

As per the docs

>- **onAvatarSelected** The callback for the avatar update result, check [AvatarUpdateResult]. Can be invoked multiple times while the Quick Editor is open.
>- **onDismiss** The callback for the dismiss action. [GravatarQuickEditorError] will be non-null if the dismiss was caused by an error.


Let me know what you think

Reply:
Thanks for the changes! 

I'm not sure about the `error` as a parameter on the `onDismiss` callback. I have some doubts about how I will handle the following case:

1. I tap in an avatar in the `GravatarQuickEditorPage`
2. The avatar is updated, so we'll receive the `onAvatarSelected` callback
3. For some reason, I lose my internet connection (any other error can replace this)
4. The `GravatarQuickEditorPage` is still open, so I select a different avatar
- Are we going to dismiss the `GravatarQuickEditorPage`? (I would say we shouldn't)
- If we don't dismiss the `GravatarQuickEditorPage`, how do we inform the app? (In case we want to notify the error. I think we should)

Maybe we should have a callback to inform the app of both the success and failure of the avatar update process and another callback to inform about the dismissal of the page/bottom sheet. Wdyt?

Reply:
Great example! 

I think in your example the error should be handled completely by us and we should properly inform the user about a network problem when selecting/uploading an avatar (the end user, not the app). I don't see a lot of value in sending those errors to the app, because there's nothing they can do about it. 

There's another group of errors though. Those that we can't handle ourselves e.g. wrong OAuth params. The app should be notified about those errors and handle properly. 

So to sum up, we have two groups (how I see it:
- errors happening during the flow related to our Gravatar service
- errors terminating the flow completely because there's not much we can do about it

Maybe the naming is wrong and the `onDismiss` param should be called `dismissReason` not an error.

Reply:
> Maybe the naming is wrong and the onDismiss param should be called dismissReason not an error.

Great idea. I think `dismissReason` is more understandable for those unsolvable flows.

> I don't see a lot of value in sending those errors to the app, because there's nothing they can do about it.

That's a fair point. It is true that in the majority of the cases, they probably don't want to do anything. However, I wonder if there could be cases where the app wants to update its UI. 

For example, in the PocketCast app, we may want to show a loading state over the avatar while we are updating or uploading the new avatar. At the moment, we are not notifying the app about it, but we would like to do it at some point. Is it possible that the app decides to finish the flow after a network error?

I may be overthinking this, but I guess it is better now rather than when we have to deal with API changes 😅

Reply:
>I may be overthinking this, but I guess it is better now rather than when we have to deal with API changes 😅

Definitely! 

> For example, in the PocketCast app, we may want to show a loading state over the avatar while we are updating or uploading the new avatar. At the moment, we are not notifying the app about it, but we would like to do it at some point. Is it possible that the app decides to finish the flow after a network error?

hmm I still think we would want to keep the user within the QuickEditor while uploading. Uploading != Selecting - as per the web flow. So the loading UI will be handled in the QE itself, there's no need for third-party apps to show it as well.

Reply:
I was thinking of a screen like:
<img width="200" alt="Alt Text" src="https://github.com/user-attachments/assets/bbd69995-8309-41fd-b327-b37a6c02cb9b">

I may be interested in using the loading state in the avatar behind the bottomsheet (App context) while the avatar is being updated. But yeah, maybe you are right, and that's too much.

Reply:
>I may be interested in using the loading state in the avatar behind the bottomsheet (App context) while the avatar is being updated. But yeah, maybe you are right, and that's too much.


Maybe! :D But again Uploading is not selecting so there's no point in showing the loading there (in the app) when it would switch back to the same avatar.

Reply:
>Do you want to update the param name? I'll do another quick round before approval. :)

yes!

Reply:
PR updated